tests/test_apigw.py

```diff
@@ -429,24 +429,144 @@ def test_wafv2(self):
             {
                 "name": "wafv2-apigw",
                 "resource": "rest-stage",
-                "filters": [{"type": "wafv2-enabled", "web-acl": "test", "state": False}],
-                "actions": [{"type": "set-wafv2", "web-acl": "test", "state": True}],
+                "filters": [{"type": "wafv2-enabled", "state": False}],
+                "actions": [{"type": "set-wafv2", "web-acl": "testv2", "state": True}],
             },
             session_factory=factory,
         )
         resources = p.run()
-        self.assertEqual(len(resources), 1)
+        self.assertEqual(len(resources), 2)
 
         p = self.load_policy(
             {
                 "name": "wafv2-apigw",
                 "resource": "rest-stage",
-                "filters": [{"type": "wafv2-enabled", "web-acl": "test", "state": True}],
+                "filters": [{"type": "wafv2-enabled", "web-acl": "testv2", "state": False}],
+                "actions": [{"type": "set-wafv2", "web-acl": "testv2", "state": True}],
             },
             session_factory=factory,
         )
+        resources = p.run()
+        self.assertEqual(len(resources), 2)
+
+        p = self.load_policy(
+            {
+                "name": "wafv2-apigw",
+                "resource": "rest-stage",
+                "filters": [{"type": "wafv2-enabled", "web-acl": "testv2", "state": True}],
+            },
+            session_factory=factory,
+        )
+        self.assertEqual(len(resources), 2)
+
+    def test_waf(self):
+        factory = self.replay_flight_data("test_rest_stage_waf")
+        p = self.load_policy(
+            {
+                "name": "waf-apigw",
+                "resource": "rest-stage",
+                "filters": [{"type": "waf-enabled", "state": False}],
+                "actions": [{"type": "set-waf", "web-acl": "test", "state": True}],
+            },
+            session_factory=factory,
+        )
+        resources = p.run()
         self.assertEqual(len(resources), 1)
 
+        p = self.load_policy(
```

Comment:
we have a strong preference for an individual test to just execute and validate one policy, and use separate tests instead of chaining policies in a single test. chaining policies obscures actual intent, and complicates refactoring as its a bit fragile.

Reply:
Updated unit tests not to chain policies